lld/ELF/Writer.cpp

```diff
@@ -2405,6 +2405,13 @@ Writer<ELFT>::createPhdrs(Partition &part) {
     addHdr(PT_GNU_STACK, perm)->p_memsz = ctx.arg.zStackSize;
   }
 
+  if (ctx.arg.osabi == ELFOSABI_OPENBSD) {
```

Comment:
The osabi check can be deleted as zNoBtCfi cannot be true for other osabi values